### PR TITLE
Add x64 support

### PR DESCRIPTION
**Pull request by Paxxi**
Adds x64 support found in the licecap library
Adds a startup hang fix by afedchin
Adds a dependency on nasm and the build system bits to download it automatically.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,7 @@ project(visualization.milkdrop2)
 cmake_minimum_required(VERSION 2.6)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR})
+set(CMAKE_ASM_NASM_COMPILER ${ADDON_DEPENDS_PATH}/bin/nasm.exe)
```
Comment (Rechi) [inline CMakeLists.txt]: `find_program`

File: depends/windowsstore/nasmw/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 2.8)
```
Comment (Rechi) [inline depends/windowsstore/nasmw/CMakeLists.txt]: why duplicate?

Comment (afedchin) [reply depends/windowsstore/nasmw/CMakeLists.txt]: no need extra dependencies folder for windows store the `windows` is used for both windows and windows store if `windowsstore` doesn't exist.

**Comment by afedchin** [comment]
after quick test I've found that not all presets work well as on old version

File: CMakeLists.txt (hunk 1 above)
Comment (Paxxi) [reply CMakeLists.txt]: I wasn't sure if find_program would set the correct variable as that one is used by enable_language(ASM_NASM)

File: depends/windowsstore/nasmw/CMakeLists.txt (hunk 2 above)
Comment (Paxxi) [reply depends/windowsstore/nasmw/CMakeLists.txt]: Will remove windows store

File: lib/vis_milk2/CMakeLists.txt [hunk 3]
```diff
@@ -56,9 +71,21 @@ set(SOURCES dx11/CommonStates.cpp
...
             vis_milk2/utility.cpp)
 
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+  set(SOURCES ${SOURCES} 
+    ns-eel2/asm-nseel-x64.asm
+    ns-eel2/glue_x86_64.h
+  )
+else()
+  set(SOURCES ${SOURCES} 
+    ns-eel2/asm-nseel-x86-msvc.c
```
Comment (afedchin) [inline lib/vis_milk2/CMakeLists.txt]: don't add `asm-nseel-x86-msvc.c` to sources. this breaks win32 build. 
`asm-nseel-x86-msvc.c` is included by `nseel-cfunc.c`

File: lib/vis_milk2/ns-eel2/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,29 @@
+project(nseel LANGUAGES C ASM_NASM)
```
Comment (afedchin) [inline lib/vis_milk2/ns-eel2/CMakeLists.txt]: what this file for?

File: lib/vis_milk2/ns-eel2/Makefile [hunk 5]
```diff
@@ -0,0 +1,120 @@
+CC=gcc
```
Comment (afedchin) [inline lib/vis_milk2/ns-eel2/Makefile]: please drop this, it's not used and will never be

File: lib/vis_milk2/ns-eel2/eel2.l [hunk 6]
```diff
@@ -0,0 +1,113 @@
+%option reentrant
```
Comment (afedchin) [inline lib/vis_milk2/ns-eel2/eel2.l]: is this used somewhere?

File: lib/vis_milk2/ns-eel2/eel2.vim [hunk 7]
```diff
@@ -0,0 +1,370 @@
+" Vim syntax file
```
Comment (afedchin) [inline lib/vis_milk2/ns-eel2/eel2.vim]: is this used somewhere?

File: lib/vis_milk2/ns-eel2/eel2.y [hunk 8]
```diff
@@ -0,0 +1,376 @@
+%pure-parser
```
Comment (afedchin) [inline lib/vis_milk2/ns-eel2/eel2.y]: is this used somewhere?

File: lib/vis_milk2/ns-eel2/eel_disasm.cpp [hunk 9]
```diff
@@ -0,0 +1,77 @@
+#include <stdio.h>
```
Comment (afedchin) [inline lib/vis_milk2/ns-eel2/eel_disasm.cpp]: is this used somewhere?

File: lib/vis_milk2/ns-eel2/gen-lex-yacc [hunk 10]
```diff
@@ -0,0 +1 @@
+yacc -v -d eel2.y && flex eel2.l
```
Comment (afedchin) [inline lib/vis_milk2/ns-eel2/gen-lex-yacc]: what this file use for?

File: lib/vis_milk2/ns-eel2/loose_eel.cpp [hunk 11]
```diff
@@ -0,0 +1,156 @@
+#include <stdio.h>
```
Comment (afedchin) [inline lib/vis_milk2/ns-eel2/loose_eel.cpp]: seems this isn't used, please drop it

File: lib/vis_milk2/ns-eel2/makefile.vc [hunk 12]
```diff
@@ -0,0 +1,45 @@
+
```
Comment (afedchin) [inline lib/vis_milk2/ns-eel2/makefile.vc]: drop this.

File: lib/vis_milk2/ns-eel2/scripts/circle.eel [hunk 13]
```diff
@@ -0,0 +1,50 @@
+gfx_init("vis",1024,768);
```
Comment (afedchin) [inline lib/vis_milk2/ns-eel2/scripts/circle.eel]: I thinks `scripts` folder also can be dropped.

File: lib/vis_milk2/ns-eel2/CMakeLists.txt (hunk 4 above)
Comment (Paxxi) [reply lib/vis_milk2/ns-eel2/CMakeLists.txt]: I used it to build only nseel when testing. Isn't really needed for anything and can be removed.

File: lib/vis_milk2/ns-eel2/Makefile (hunk 5 above)
Comment (Paxxi) [reply lib/vis_milk2/ns-eel2/Makefile]: the Makefile isn't used at all, only included it as it makes diffing easier if there's ever any upstream work we're interested in.

File: lib/vis_milk2/ns-eel2/eel2.l (hunk 6 above)
Comment (Paxxi) [reply lib/vis_milk2/ns-eel2/eel2.l]: no

File: lib/vis_milk2/ns-eel2/eel2.vim (hunk 7 above)
Comment (Paxxi) [reply lib/vis_milk2/ns-eel2/eel2.vim]: no

File: lib/vis_milk2/ns-eel2/eel2.y (hunk 8 above)
Comment (Paxxi) [reply lib/vis_milk2/ns-eel2/eel2.y]: no

File: lib/vis_milk2/ns-eel2/eel_disasm.cpp (hunk 9 above)
Comment (Paxxi) [reply lib/vis_milk2/ns-eel2/eel_disasm.cpp]: no

File: lib/vis_milk2/ns-eel2/gen-lex-yacc (hunk 10 above)
Comment (Paxxi) [reply lib/vis_milk2/ns-eel2/gen-lex-yacc]: yacc/lex/bison files are used to generate the parser. They aren't used, not sure if they're of any value.

File: lib/vis_milk2/ns-eel2/loose_eel.cpp (hunk 11 above)
Comment (Paxxi) [reply lib/vis_milk2/ns-eel2/loose_eel.cpp]: yep, this is to build some tool, same with eel_disasm

File: lib/vis_milk2/ns-eel2/makefile.vc (hunk 12 above)
Comment (Paxxi) [reply lib/vis_milk2/ns-eel2/makefile.vc]: yep

File: lib/vis_milk2/ns-eel2/scripts/circle.eel (hunk 13 above)
Comment (Paxxi) [reply lib/vis_milk2/ns-eel2/scripts/circle.eel]: yep

File: lib/vis_milk2/CMakeLists.txt (hunk 3 above)
Comment (Paxxi) [reply lib/vis_milk2/CMakeLists.txt]: ahh, didn't notice that

File: CMakeLists.txt [hunk 14]
```diff
@@ -4,6 +4,9 @@ cmake_minimum_required(VERSION 2.6)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR})
 
+find_program(CMAKE_ASM_NASM_COMPILER nasm
+  PATHS ${ADDON_DEPENDS_PATH}/bin)
```
Comment (Rechi) [inline CMakeLists.txt:8]: no need to specify a path here, that's the default one

**Comment by afedchin** [comment]
thanks for wasted time on this but working patch went in

**Pull request closed without merging**